tiledb/sm/filter/filter.cc

```diff
@@ -63,34 +64,44 @@ Filter* Filter::clone() const {
   return clone;
 }
 
-Filter* Filter::create(FilterType type) {
+std::tuple<Status, std::optional<std::shared_ptr<Filter>>> Filter::create(
+    FilterType type) {
```

Comment:
I'd recommend not changing this function in the present PR. It's only called in the C API, where there's no notion of C.41 compliance, so there's no reason to use new constructors. The C API needs to be dealt with separately, and it would be prudent not to incur any risk of change until we're ready to deal with it all at once.